lib/bot.js

```diff
@@ -106,44 +88,47 @@ class Bot {
     return this._safelyRunApiCommand({method: 'read', options: {channel, peek, unreadOnly}})
   }
 
-  // --------------------------------------------------------------------------
-
   chatDelete(options: {channel: ChatChannel, messageId: number}): Promise<any> {
     const {channel, messageId} = options
     return this._safelyRunApiCommand({method: 'delete', options: {channel, messageId}})
   }
 
-  // --------------------------------------------------------------------------
-
   myInfo(): ?DeviceUsernamePair {
     return this._dPair
   }
 
-  // --------------------------------------------------------------------------
-
-  watchChannelForNewMessages(options: {channel: ChatChannel, onMessages: MessagesHandler}) {
-    const {channel, onMessages} = options
-    const key = chatChannelToKey(channel)
-    if (this._channelWatchers.has(key)) {
-      throw new Error(`already watching the channel ${JSON.stringify(channel)}`)
-    } else {
-      this._channelWatchers.set(key, new ChannelWatcher({channel, onMessages, bot: this}))
-    }
+  /**
+   * Listens for new chat messages on a specified channel. The `onMessage` function is called for every message your bot receives.
+   * @param onMessage
+   * @memberof Bot
+   */
+  watchChannelForNewMessages(channel: ChatChannel, onMessage: (message: ChatReadMessage) => void) {
+    keybaseExec(['chat', 'api-listen'], {
+      onStdOut: chunk => {
+        const streamString = chunk.toString('utf8')
+        const messageObject = JSON.parse(streamString)
```

Comment:
Is it safe to assume that the `chunk` size will be the full JSON text? What if the JSON is larger than the chunk size?

The number of bytes read is defaulted to 16Kb for Readable streams: https://nodejs.org/api/stream.html#stream_new_stream_readable_options

But that depends on how `spawn` initializes the `Stream.Readable` for `STDOUT`.

This might not be an issue in testing, but I'm curious if it's possible to get a `JSON.parse` error, which reminds me that it should be wrapped in `try { ... } catch (e) { ... }`